File: codingTest/baekjoon/javascript/9012.js
const rl = require("readline").createInterface({ input: process.stdin, output: process.stdout });

const input = [];
rl.on("line", line => {
  input.push(line);
}).on("close", () => {
  input.shift();
  for (let s of input) {
    let stack = 0;
    for (let c of s) {
      if (stack < 0) break;
      c == "(" ? stack++ : stack--;
    }
    if (stack == 0) console.log("YES");
    else console.log("NO");
  }
});

File: codingTest/baekjoon/python/9012.py
import sys
input = sys.stdin.readline

def yesOrNo(s):
    stack = []
    for i in s:
        if i == "(": stack.append(i)
        else:
            if len(stack) == 0: return "NO"
            else: stack.pop()
    return "YES" if len(stack) == 0 else "NO"

for _ in range(int(input())):
    s = input().rstrip()
    if s[0] == ")": print("NO")
    else: print(yesOrNo(s))